addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/thirdgeneration/ThirdGenerationHandler.java

```diff
@@ -0,0 +1,492 @@
+import javax.measure.quantity.Time;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link ThirdGenerationHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author René Stakemeier - Initial contribution
+ */
+@NonNullByDefault
+public class ThirdGenerationHandler extends BaseThingHandler {
+
+    /*
+     * operations used for authentication
+     */
+    private static final String AUTH_START = "/auth/start";
+    private static final String AUTH_FINISH = "/auth/finish";
+    private static final String AUTH_CREATE_SESSION = "/auth/create_session";
+
+    /*
+     * operations used for gathering process data from the device
+     */
+    private static final String PROCESSDATA = "/processdata";
+
+    /*
+     * After the authentication the result (the session id) is stored here and used to "sign" future requests
+     */
+    private @Nullable String sessionId;
+    /*
+     * The configuration file containing the host, the password and the refresh interval
+     */
+    private @NonNullByDefault({}) ThirdGenerationConfiguration config;
+
+    private @Nullable ScheduledFuture<?> refreshScheduler;
+
+    private @Nullable HttpClient httpClient;
+
+    private ThirdGenerationInverterTypes inverterType;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * Constructor of this class
+     *
+     * @param thing        the thing
+     * @param httpClient   the httpClient used for communication
+     * @param inverterType the type of the device
+     */
+    public ThirdGenerationHandler(Thing thing, HttpClient httpClient, ThirdGenerationInverterTypes inverterType) {
+        super(thing);
+        this.inverterType = inverterType;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // All channels are readonly and updated by the scheduler
+    }
+
+    @Override
+    public void dispose() {
+        if (refreshScheduler != null) {
+            refreshScheduler.cancel(true);
```

Comment:
You need to assign `null` to refreshScheduler otherwise it won't be cancelled the next time or check also for is cancelled.

Reply:
This part of the code was generated by the `create_openhab_binding_skeleton.cmd`

From my understanding the dispose method is only called in case the object is destroyed, so there won´t be "the next time".

Or is there anything I´m missing here? :-)

Reply:
Every time a configuration is changed the handler will do a dispose/initialize cycle. Also if the user disables/enables a thing.